hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/CyclicWriteBehindQueue.java

```diff
@@ -180,13 +181,13 @@ public void getFrontByNumber(int numberOfElements, Collection<DelayedEntry> coll
 
     private void addCountIndex(DelayedEntry entry) {
         Data key = (Data) entry.getKey();
-        Map<Data, Integer> index = this.index;
+        Map<Data, MutableInteger> index = this.index;
 
-        Integer count = index.get(key);
+        MutableInteger count = index.get(key);
         if (count == null) {
-            count = 0;
+            count = new MutableInteger();
         }
-        count++;
+        count.value++;
```

Comment:
what about encapsulating the Map + MutableInteger in something like
```java
public class Counter<T> {
  int incAndGet(T data) {
    [...]
  }  

  int decAntGet(T data) {
    [...]
  }

  [ ...other methods... ]
}
```

this would allow you to remove implementation noise such as
```java
        MutableInteger count = index.get(key);
        if (count == null) {
            count = new MutableInteger();
        }
```
from the CyclicWriteBehindQueue class